tests/unit/test_hnsw_tiered.cpp

```diff
@@ -78,11 +74,94 @@ TYPED_TEST(HNSWTieredIndexTest, CreateIndexInstance) {
         ASSERT_EQ(tiered_index->getDistanceFrom(1, vector), 0);
         ASSERT_EQ(memory_ctx, tiered_index->getAllocator()->getAllocationSize());
         ASSERT_EQ(tiered_index->flatBuffer->indexSize(), 0);
-        ASSERT_EQ(tiered_index->labelToInsertJobs[vector_label].size(), 0);
+        ASSERT_EQ(tiered_index->labelToInsertJobs.at(vector_label).size(), 0);
+
+        // Cleanup.
+        delete jobQ;
+        VecSimIndex_Free(tiered_index);
+    }
+}
+
+TYPED_TEST(HNSWTieredIndexTest, addVector) {
```

Comment:
Nice test! but don't you want to validate the index state after performing the insertion job?

Reply:
Implementing the insertion background job is not part of this PR (but it's coming... very soon...)